src/dmd/hdrgen.d

```diff
@@ -1281,7 +1281,11 @@ public:
     {
         protectionToBuffer(buf, d.protection);
         buf.writeByte(' ');
-        visit(cast(AttribDeclaration)d);
+        AttribDeclaration ad = cast(AttribDeclaration)d;
+        if (ad.decl.dim == 1 && strcmp((*ad.decl)[0].kind, "protection attribute") == 0)
```

Comment:
Is there no enum or other way to compare rather than comparing strings?

Reply:
Nope. I looked for a `isProtDeclaration()` function, but there is no such thing.

Reply:
How about adding that function? Or can you visit `(*ad.decl)[0]`?

Reply:
Visiting is not a possibility since that will make it visit ProtDeclaration again and print `private public`. I will add the function.

Reply:
Is this ok ?

Reply:
you should be able to check the tok

Reply:
@stefan-koch-sociomantic there's no token associated with a `AttribDeclaration`. It's only expressions that stores a token. I would be nice if all AST nodes store a kind or type, similar to the token in expression. So it's possible to use the same way for all AST nodes to identify subclasses.